src/core/IO/Socket/INET.pm

```diff
@@ -55,18 +53,34 @@ my class IO::Socket::INET does IO::Socket {
         return $host ?? ($host, $port) !! $uri;
     }
 
+    method !correct-family(Int:D $family) {
+        if ($family == 3) {
+            note "You're using the old value (3) for IPv6 / PF_NET6, " ~
+                "please use the constant PF_INET6 to specify IPv6";
+            note Backtrace.new.Str;
+
```

Comment:
Here, would you change to use [DEPRECATED() routine](https://github.com/rakudo/rakudo/blob/ebd6440c27ea07e2bfde4809fa4e97dac1491e69/src/core/Deprecations.pm#L54) instead of two `note`s. If you grep the codebase for `DEPRECATED` you'll find a few usage examples.

Reply:
I want to use the DEPRECATED routine, but I've been unable to get DEPRECATED to skip the last 2 functions in the trace (as those are just internal functions).

any tips how to get around that?

Reply:
I think that's the `:up` argument. Try giving it some numbers. Grep the source for other uses of DEPRECATED and see how it's used.

Reply:
yes, I've been unable to get that to work, would always keep using the last call (`!correct-family`), even if I put `:up` on a ridiculous high number, maybe it has been fixed now, but this is why this this PR didn't progress after that anymore.

Reply:
And `:up(*)` doesn't work either?

Reply:
Didn't try that yet! will do when I have some time on my hands, thanks @zoffixznet

Reply:
Works! thanks @zoffixznet!